[js] Add a guide for downloading spec (#1128)
SAP/cloud-sdk: docs-js/guides/apihub-download-spec.mdx

---
id: api-business-hub-download-specification
title: Download Specification Files from SAP API Business Hub
sidebar_label: Download specification files
description: This guide demonstrates the steps to download OData and OpenAPI specification files (specs) from SAP API Business Hub.
---

import useBaseUrl from '@docusaurus/useBaseUrl';

This guide demonstrates the steps to download OData and OpenAPI specification files (specs) from SAP API Business Hub.

## When Do I Need A Specification File?

> When you consume an OData or OpenAPI service and use the following packages, the specs are needed:

Reviewer: ```suggestion
When you want to consume an OData or OpenAPI service you need to generate a corresponding client.
For the generation you need the service definition or specs and the right generator:
```

Reviewer 2: [pp] To consume an OData or OpenAPI service, you need to generate a corresponding client.

To avoid too many 'you'.

Reviewer: yes that is better.

- `@sap-cloud-sdk/generator` for generating an OData client (see [doc](../features/odata/generate-odata-client.mdx))
- `@sap-cloud-sdk/openapi-generator` for generating an OpenAPI client (see [doc](../features/openapi/generate-openapi-client.mdx))

## Steps To Download A Specification File

1. Go to [SAP API Business Hub](https://api.sap.com).

> 2. Search for the service you want to consume, and you should see the search result like the screenshot below.

Reviewer: [pp] This sentence does not read so smooth Perhaps split it in two?

3. Click login, as this is necessary for downloading the specs.
4. You can use filters like "SAP S/4HANA Cloud" for targeting your service.

<img src={useBaseUrl('img/apihub-download-spec-1.png')} />

5. Click the service name, when you find your service, so you can see a similar page like the screenshot below.

> 6. Click the "API Specification"

Reviewer 2: [req] Click **API Specification**
bold the text -> https://github.com/SAP/cloud-sdk/blob/main/STYLEGUIDE.md#use-bold-format-for-names

[pp] Image 2 and 3 both highlight the API Specification (with the red arrow), so I would just combine the sentence and add only 1 image. So the sentence would read:
Click **API Specification** and download EDMX file (For OpenAPI, choose YAML or JSON)

Author: > so I would just combine the sentence and add only 1 image.

The idea is to show the **API Specification** and the BuPa service name in the first image.
Then the screen is not big enough, so I introduce the second one.

   <img src={useBaseUrl('img/apihub-download-spec-2.png')} />

7. Scroll down and download the "EDMX" file for an OData service. (For OpenAPI, choose YAML or JSON)
   <img src={useBaseUrl('img/apihub-download-spec-3.png')} />

:::tip Try out sandbox systems

> Sometimes, you are not 100% sure whether you choose the right service.

Reviewer 2: [pp] Shorten the sentence + active voice

If you're unsure whether the chosen service is correct, click **Try Out** to test the sandbox system, which exposes a few endpoints.

## When Do I Need A Specification File?

To consume an OData or OpenAPI service, you need to generate a corresponding client.
The documentation of the generator packages can be found below:

- `@sap-cloud-sdk/generator` for generating an OData client (see [doc](../features/odata/generate-odata-client.mdx))
- `@sap-cloud-sdk/openapi-generator` for generating an OpenAPI client (see [doc](../features/openapi/generate-openapi-client.mdx))

## Steps To Download A Specification File

1. Go to [SAP API Business Hub](https://api.sap.com).

> 2. Click login, as this is necessary for downloading the specs.

Reviewer 2: ```suggestion
2. Click **login**, as this is necessary for downloading the specs.
```
UI element in bold

3. Search for the service you want to consume like "business partner". The search results should look like the screenshot below.
4. Apply filters for targeting a specific service by clicking the **Show Filters** button. For example, to filter only S/4HANA Cloud services, in the `Product` list, select `SAP S/4HANA Cloud`.

<img src={useBaseUrl('img/apihub-download-spec-1.png')} />

5. Click the service name, when you find your service, so you can see a similar page like the screenshot below.

> 6. Click the **API Specification**

Reviewer 2: ```suggestion
6. Click **API Specification**
```

   <img src={useBaseUrl('img/apihub-download-spec-2.png')} />

> 7. Scroll down and download the "EDMX" file for an OData service. (For OpenAPI, choose YAML or JSON)

Reviewer 2: ```suggestion
7. Scroll down and download the EDMX file for an OData service. (For OpenAPI, choose YAML or JSON)
```
